pkg/imgpkg/registry/keychain.go

```diff
@@ -14,6 +15,10 @@ import (
 // keychains that contain credentials for 'any' target. i.e. env keychain takes precedence over the custom keychain.
 // Since env keychain contains credentials per HOSTNAME, and custom keychain doesn't.
 func Keychain(keychainOpts auth.KeychainOpts, environFunc func() []string) (regauthn.Keychain, error) {
+	if environFunc == nil {
+		environFunc = os.Environ
+	}
```

Comment:
should this be pushed down to EnvKeychain?

Reply:
Created a builder for the EnvKeychain that takes care of this, good catch